### ISSUE TITLE
Maestro generated PRs for dependency updates have incorrect commit diff links

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/arcade#7339]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

I noticed from https://github.com/dotnet/runtime/pull/51981 that the PR description's links to the diffs all point to the same URL. The very first rendered diff link from xharness is used again for all other diff links throughout the list.

I presume the expected behavior is that each link should go to the source repo from the update and show a diff between the two commits for the specific update.

**Comment by riarenas** [comment]
Looks like a bug in this space that affects batched subscriptions. It seems like the only link that is written to the PR is whichever update was the first to create the PR.

**Comment by riarenas** [comment]
This can potentially be considered FR work as although it's not blocking, it's causing customer confusion.

**Comment by jakubstilec** [comment]
[Async Triage]: In our opinion this is a client facing bug and should be done as part of FR.

**Comment by riarenas** [comment]
I don't think it's FR from either the "blocking" definition as it's a missed feature. Maybe FR could do an investigation on what would be needed to implement and if it fits the size criteria for FR issues it would go there, and otherwise #1476 as @MattGal suggests.

**Comment by markwilkie** [comment]
adding to FR

**Comment by premun** [comment]
I haven't had time to work on this so putting back to backlog.

**Comment by jakubstilec** [comment]
The issue is in GitHub link references. Inside PR's description we have multiple references like:
```
[1]: https://github.com/dotnet/xharness/compare/0e95171...89cb4b1
[1]: https://github.com/dotnet/arcade/compare/b7279bb...0cfaf93
```

Then once you use this reference in a link, everything uses the first definition. The fix should be straightforward.

**Comment by jakubstilec** [comment]
Prepared a PR https://github.com/dotnet/arcade-services/pull/1637, but still looking how to verify this change properly.

**Comment by lpatalas** [comment]
@jakubstilec handed it over to me. The only thing remaining here is to review the PR and add some tests for it. I'm working on it.

**Comment by lpatalas** [comment]
Due to other FR issues I didn't have time to move it forward. So the status did not change from what I wrote above. Hopefully I will be able to get back to it before end of the week.

**Comment by lpatalas** [comment]
We had discussion with @ilyas1974 and @jakubstilec about this issue because it is stuck in FR for quite some time due to other more urgent issues. Jakub did first part of the work in the [PR](https://github.com/dotnet/arcade-services/pull/1637). The part that remains is writing tests for the functionality. Unfortunately it's not as straightforward as it should be. Some problems are:
- The implementation is not split into separate components so it's hard to test them in isolation
- We only have end-to-end tests for this code in [PullRequestActorTests.cs](https://github.com/dotnet/arcade-services/blob/main/src/Maestro/tests/SubscriptionActorService.Tests/PullRequestActorTests.cs)
- existing tests are hardcoded to support only single build. There's no easy way to set them up in such a way that we have multiple diff links which is what we are interested in this issue
- we could just push on add new tests in the same way but that means we can be in the same situation next time we are working in this area
- it's hard to test the functionality on local machine

In other words the code requires some refactoring to make it more testable and easier to work with. Because of that we agreed that the best way forward is to move this issue outside of FR and triage it. 

The arguments for that are:
- This issue is present on production since at least half a year ago which suggests that it's not critical
- It will be a better long term investment to refactor this code and make our life easier rather than doing a quick fix as part of FR
- We can put more time into removing tech debt and making the code testable when not constrained by limited resources we have during FR which we can put into solving more urgent issues

**Comment by adiaaida** [comment]
Is this actually in PR? Should we move it back to the backlog?

**Comment by ilyas1974** [comment]
[Async-Triage]: In reviewing this with the Prague team, the issue is still not resolved for the customer because we do not have a way (currently) of validating our changes. @dkurepa is going to take a look at this issue again.

**Comment by dkurepa** [comment]
Here's the link for the PR: https://github.com/dotnet/arcade-services/pull/1774

**Comment by lpatalas** [comment]
@dkurepa I will move it into rollout column because it's already merged.

**Comment by dkurepa** [comment]
We used this PR to test the changes, and it looks good.
https://github.com/dotnet/runtime/pull/62279
At first there was only one dependency. Then we had an update which created the second dependency and it had the correct link. Finally, there was another update which affected the second dependency and correctly updated the link, confirming both use cases.

**Comment by premun** [comment]
This has been fixed, closing
